# Converting Significance Level (α) to Z-Score and Vice Versa Using SciPy

(written mostly by Grok)
This notebook demonstrates how to convert between a significance level (α) and a Z-score using Python's `scipy.stats.norm` module, which provides functions for the standard normal distribution. We include formulas, Python code, and explanations for both conversions, covering one-tailed and two-tailed tests.


## 1. Significance Level to Z-Score

The significance level (α) represents the probability in the tails of the standard normal distribution (e.g., α = 0.05 for a 5% significance level). To find the Z-score, we use the inverse cumulative distribution function (percent point function, `ppf`).

### Formulas

**Two-tailed test**:
- α is split equally between the two tails (α/2 in each tail).
- Z-score is found such that the area to the right of Z (or left of -Z) is α/2:
  $$
  Z = \Phi^{-1}(1 - \alpha/2)
  $$
  where $\Phi^{-1}$ is the inverse of the standard normal CDF.

**One-tailed test (right tail)**:
- Z-score is found such that the area to the right of Z is α:
  $$
  Z = \Phi^{-1}(1 - \alpha)
  $$

**One-tailed test (left tail)**:
- Z-score is found such that the area to the left of Z is α:
  $$
  Z = \Phi^{-1}(\alpha)
  $$

### Python Code
Below, we compute Z-scores for a significance level of α = 0.05.

In [1]:
from scipy.stats import norm

# Significance level
alpha = 0.05

# Two-tailed test: Z-score for alpha/2 in each tail
z_two_tailed = norm.ppf(1 - alpha / 2)
print(f'Two-tailed Z-score for α = {alpha}: {z_two_tailed:.4f}')

# One-tailed test (right tail): Z-score for alpha
z_one_tailed_right = norm.ppf(1 - alpha)
print(f'One-tailed (right) Z-score for α = {alpha}: {z_one_tailed_right:.4f}')

# One-tailed test (left tail): Z-score for alpha
z_one_tailed_left = norm.ppf(alpha)
print(f'One-tailed (left) Z-score for α = {alpha}: {z_one_tailed_left:.4f}')

Two-tailed Z-score for α = 0.05: 1.9600
One-tailed (right) Z-score for α = 0.05: 1.6449
One-tailed (left) Z-score for α = 0.05: -1.6449


### Explanation of Output
- **Two-tailed**: For α = 0.05, the Z-score is approximately 1.9600, meaning each tail (α/2 = 0.025) corresponds to a Z-score of ±1.96.
- **One-tailed (right)**: The Z-score is approximately 1.6449, where the right tail has area α = 0.05.
- **One-tailed (left)**: The Z-score is approximately -1.6449, where the left tail has area α = 0.05.

## 2. Z-Score to Significance Level

To convert a Z-score to a significance level (α), we calculate the tail probability using the cumulative distribution function (`cdf`) of the standard normal distribution.

### Formulas

**Two-tailed test**:
- The total significance level is the sum of the probabilities in both tails:
  $$
  \alpha = 2 \cdot (1 - \Phi(|Z|))
  $$
  where $\Phi(Z)$ is the CDF of the standard normal distribution.

**One-tailed test (right tail)**:
- The significance level is the area to the right of Z:
  $$
  \alpha = 1 - \Phi(Z)
  $$

**One-tailed test (left tail)**:
- The significance level is the area to the left of Z:
  $$
  \alpha = \Phi(Z)
  $$

### Python Code
Below, we compute the significance level for a Z-score of 1.96.

In [2]:
from scipy.stats import norm

# Z-score
z = 1.96

# Two-tailed test: α = 2 * P(Z > |z|)
alpha_two_tailed = 2 * (1 - norm.cdf(abs(z)))
print(f'Two-tailed α for Z = {z}: {alpha_two_tailed:.4f}')

# One-tailed test (right tail): α = P(Z > z)
alpha_one_tailed_right = 1 - norm.cdf(z)
print(f'One-tailed (right) α for Z = {z}: {alpha_one_tailed_right:.4f}')

# One-tailed test (left tail): α = P(Z < z)
alpha_one_tailed_left = norm.cdf(z)
print(f'One-tailed (left) α for Z = {z}: {alpha_one_tailed_left:.4f}')

Two-tailed α for Z = 1.96: 0.0500
One-tailed (right) α for Z = 1.96: 0.0250
One-tailed (left) α for Z = 1.96: 0.9750


### Explanation of Output
- **Two-tailed**: For Z = 1.96, α is approximately 0.0500, meaning the total area in both tails is 5%.
- **One-tailed (right)**: The area to the right of Z = 1.96 is approximately 0.0250 (half of the two-tailed α).
- **One-tailed (left)**: The area to the left of Z = 1.96 is approximately 0.9750, which corresponds to the cumulative probability.

## Notes
- The `norm.ppf` function computes the Z-score for a given cumulative probability (inverse CDF).
- The `norm.cdf` function computes the cumulative probability for a given Z-score.
- For two-tailed tests, use the absolute value of the Z-score and account for both tails.
- The examples assume a standard normal distribution (mean = 0, standard deviation = 1).
- For non-standard normal distributions, adjust the mean and standard deviation in `norm.ppf` or `norm.cdf` (e.g., `norm.ppf(1 - alpha, loc=mean, scale=std)`).
